adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepositoryTest.java

```diff
@@ -447,4 +443,49 @@ void addFareRuleAndGetFareRuleShouldReturnSameEntity() {
         assertEquals(mockFareRule01, underTest.getFareRule("fare id1", null, null,
                 null, null));
     }
+
+
+    @Test
+    void callToAddShapeShouldSameReturnEntity() {
+        final ShapePoint mockShapePoint = mock(ShapePoint.class);
+        when(mockShapePoint.getShapeId()).thenReturn("test id");
+        final InMemoryGtfsDataRepository underTest = new InMemoryGtfsDataRepository();
+
+        assertEquals(mockShapePoint, underTest.addShapePoint(mockShapePoint));
+    }
+
+    @Test
+    void addTwiceSameShapePointShouldReturnNull() {
```

Comment:
this only test adding a single instance twice. We could have input data that contains twice the same row, that would be represented by two ShapePoint instances containing the same data. I suggest we add another test for that case. In our other entity classes we don't need that because checking IDs is sufficient.